### ISSUE TITLE
Support template debugging user journeys

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#121]
### TL;DR

As a template developer, I want tools to understand why my `actions` aren't having the effect I intended.

Ideas:
 - ~~Offer the ability to fail with a helpful error message if an action doesn't affect any files~~ some actions, like `print`, are working as intended when they don't affect any files.
 - ~~Offer the ability to track changes to a particular file~~ we can skip this because it's solved by the next bullet point below
 - Offer the ability to show a diff of changes to each file due to each action

Maybe some of these use cases can be met with improved debug logging? Not sure.

**Comment by sqin2019** [comment]
@drevell I am thinking two approaches for this problem:
1. use git, we can see changes for each step execution with `git log`, we are also able to check if there are any diffs or not for each step. we will end up having a `.git` folder in the destination though.
2. do not use git, in each action, we need to add a function to output the changes.

I prefer the first one, since it scales (no changes required for new actions) and clean, it does require the user to have git installed which should not be a concern as git is somehow required. 

Just quickly check with you to see if I am missing anything and what your thoughts are.
